packages/intent-filler/src/strategies/basic.ts

```diff
@@ -0,0 +1,487 @@
+		} catch (error) {
+			console.error(`Error checking token balances:`, error)
+			return false
+		}
+	}
+
+	/**
+	 * Calculates the total order value for confirmation policy
+	 */
+	private async calculateOrderValue(order: Order): Promise<BigInt> {
+		let totalUSDValue = BigInt(0)
+
+		for (const input of order.inputs) {
+			const tokenUsdPrice = await fetchTokenUsdPriceOnchain(
+				input.token,
+				this.provider,
+				addresses.UniswapV2Router[order.destChain as keyof typeof addresses.UniswapV2Router]!,
+				assets[order.destChain as keyof typeof assets].WETH,
+				assets[order.destChain as keyof typeof assets].USDC,
+			)
+
+			totalUSDValue = totalUSDValue + BigInt(input.amount * BigInt(tokenUsdPrice))
+		}
+
+		return totalUSDValue
+	}
+
+	/**
+	 * Estimates gas for filling an order
+	 */
+	private async estimateGasForFill(order: Order): Promise<BigInt> {
+		try {
+			const contractOrder = this.transformOrderForContract(order)
+
+			const fillOptions: FillOptions = {
+				relayerFee: ethers.utils.parseEther("0.001").toString(),
+			}
+
+			const ethValue = await this.calculateRequiredEthValue(order.outputs)
+
+			const gasEstimate = await this.contract.estimateGas.fillOrder(contractOrder, fillOptions, {
+				value: ethValue,
+			})
+
+			return BigInt(gasEstimate.toString())
+		} catch (error) {
+			console.error(`Error estimating gas:`, error)
+			// Return a conservative estimate if we can't calculate precisely
+			return BigInt(500000)
+		}
+	}
+
+	/**
+	 * Gets the current ETH price in USD
+	 */
+	private async getEthPriceUsd(): Promise<number> {
+		const ethPriceUsd = await fetchTokenUsdPriceOnchain(
+			assets[this.chain as keyof typeof assets].WETH,
+			this.provider,
+			addresses.UniswapV2Router[this.chain as keyof typeof addresses.UniswapV2Router]!,
+			assets[this.chain as keyof typeof assets].WETH,
+			assets[this.chain as keyof typeof assets].USDC,
+		)
+
+		return ethPriceUsd
+	}
+
+	/**
+	 * Gets the HyperBridge protocol fee in ETH
+	 */
+	private async getProtocolFeeEth(order: Order): Promise<number> {
+		const requestBody = this.constructRedeemEscrowRequest(order)
+
+		const dispatchPost: DispatchPost = {
+			dest: order.sourceChain,
+			to: addresses.IntentGateway[order.sourceChain as keyof typeof addresses.IntentGateway]!,
+			body: requestBody,
+			timeout: order.deadline,
+			fee: order.fees,
+			payer: order.user,
+		}
+
+		const protocolFeeEth = await this.contract.quote(dispatchPost)
+
+		return protocolFeeEth
+	}
+
+	/**
+	 * Constructs the redeem escrow request body
+	 */
+	private constructRedeemEscrowRequest(order: Order): HexString {
+		const commitment = getOrderCommitment(order)
+
+		// RequestKind.RedeemEscrow is 0 as defined in the contract
+		const requestKind = encodePacked(["uint8"], [RequestKind.RedeemEscrow])
+
+		const requestBody = encodePacked(
+			["bytes32", "bytes32", "tuple(bytes32 token, uint256 amount)[]"],
+			[commitment as HexString, this.wallet.address as HexString, order.inputs],
+		)
```

Comment:
feels like we should have these utility methods in the sdk and import them from there

Reply:
okay noted, I am keeping this pr comment unresolved, will move all the utils to the sdk